Geometry_Engine/Query/GeometryHash.cs

```diff
@@ -0,0 +1,446 @@
+        [Output("geomHash", "Array of numbers representing a unique signature of the input geometry.")]
+        public static double[] IGeometryHash(this IGeometry igeometry)
+        {
+            return GeometryHash(igeometry as dynamic, 0);
+        }
+
+        /***************************************************/
+        /****              Private Methods              ****/
+        /***************************************************/
+
+
+        /***************************************************/
+        /****  Curves                                   ****/
+        /***************************************************/
+
+        [Description("The geometry hash of a Curve is obtained by first retrieving any Sub-part of the curve, if present." +
+            "The ISubParts() methods is able to return the 'primitive' curves that a curve is composed of. " +
+            "The GeometryHashes are then calculated for the individual parts and concatenated.")]
+        private static double[] GeometryHash(this ICurve curve, double translationFactor)
+        {
+            List<ICurve> subParts = curve.ISubParts().ToList();
+
+            List<double> hashes = new List<double>();
+
+            //Add hash ignoring endpoint for all but last curve
+            for (int i = 0; i < subParts.Count - 1; i++)
+            {
+                hashes.AddRange(GeometryHash(subParts[i] as dynamic, translationFactor, true));
+            }
+            //Include endpoint for hasing for last curve
+            hashes.AddRange(GeometryHash(subParts.Last() as dynamic, translationFactor, false));
+
+            return hashes.ToArray();
+        }
+
+        /***************************************************/
+
+        [Description("The GeometryHash for an Arc is calculated as the GeometryHash of the start, end and middle point of the Arc.")]
+        private static double[] GeometryHash(this Arc curve, double translationFactor, bool skipEndPoint= false)
+        {
+            translationFactor += (int)TypeTranslationFactor.Arc;
+
+            IEnumerable<double> hash = curve.StartPoint().ToDoubleArray(translationFactor)
+               .Concat(curve.PointAtParameter(0.5).ToDoubleArray(translationFactor));
+
+            if (!skipEndPoint)
+                hash = hash.Concat(curve.EndPoint().ToDoubleArray(translationFactor));
+
+            return hash.ToArray();
+        }
+
+        /***************************************************/
+
+        [Description("The GeometryHash for an Circle is calculated as the GeometryHash of the start, 1/3rd and 2/3rd points of the Circle.")]
+        private static double[] GeometryHash(this Circle curve, double translationFactor, bool skipEndPoint = false)
+        {
+            // The input `skipEndPoint` is not used here because Ellipses cannot be part of Polycurves.
```

Comment:
This is actually not correct. Circles and Ellipses can be part of PolyCurves.

The reason for not skipping end point (the way I look at it) is that ellipses do not have clear start and endpoints that could be seen as segments along a continuous curve. Would change comment to something like:
```suggestion
            // The input `skipEndPoint` is not used here because Circles do not have a clearly defined endpoint to be used in a chain of segment curves.
```
or something to that avail.